catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostIngestPlugin.java

```diff
@@ -115,63 +118,104 @@
-    return createStorageRequest;
+    updateContent(
+        createResponse.getCreatedMetacards(), createResponse.getRequest().getProperties());
+    return createResponse;
   }
 
   @Override
-  public UpdateStorageRequest process(UpdateStorageRequest updateStorageRequest)
-      throws PluginExecutionException {
-    if (updateStorageRequest == null) {
-      throw new PluginExecutionException(
-          "process(): argument 'updateStorageRequest' may not be null.");
+  public UpdateResponse process(UpdateResponse updateResponse) throws PluginExecutionException {
+    if (updateResponse == null) {
+      throw new PluginExecutionException("process(): argument 'updateResponse' may not be null.");
     }
+    updateContent(
+        updateResponse
+            .getUpdatedMetacards()
+            .stream()
+            .map(Update::getNewMetacard)
+            .collect(Collectors.toList()),
+        updateResponse.getRequest().getProperties());
+    return updateResponse;
+  }
 
-    process(updateStorageRequest.getContentItems());
-    return updateStorageRequest;
+  @Override
+  public DeleteResponse process(DeleteResponse deleteResponse) throws PluginExecutionException {
+    return deleteResponse;
   }
 
-  private boolean isNitfMimeType(String rawMimeType) {
-    try {
-      return NITF_MIME_TYPE.match(rawMimeType);
-    } catch (MimeTypeParseException e) {
-      LOGGER.debug("unable to compare mime types: {} vs {}", NITF_MIME_TYPE, rawMimeType);
+  private void updateContent(List<Metacard> metacards, Map<String, Serializable> properties) {
+    for (Metacard mcard : metacards) {
+      if (shouldGenerateContentItems(mcard, properties)) {
+        try {
+          createContentItems(mcard);
+        } catch (Exception e) {
+          LOGGER.warn("Error creating thumbnail/overview/original for {}", mcard.getId());
+        }
+      }
     }
+  }
 
-    return false;
+  private boolean shouldGenerateContentItems(
+      Metacard metacard, Map<String, Serializable> properties) {
+    Attribute type = metacard.getAttribute(Media.TYPE);
+    return type != null
+        && IMAGE_NITF.equals(type.getValue())
+        && !(boolean) properties.getOrDefault(NITF_PROCESSING_KEY, false);
   }
 
-  private void process(List<ContentItem> contentItems) {
+  private void createContentItems(Metacard metacard) throws IOException {
+    ResourceResponse response;
+    try {
+      response = catalogFramework.getLocalResource(new ResourceRequestById(metacard.getId()));
+    } catch (ResourceNotFoundException | ResourceNotSupportedException | IOException e) {
+      LOGGER.warn("Error retrieving resource for thumbnail creation", e);
+      return;
+    }
+
+    ByteSource byteSource =
+        new ByteSource() {
+          @Override
+          public InputStream openStream() throws IOException {
+            return response.getResource().getInputStream();
+          }
+        };
+    ContentItem contentItem =
```

Comment:
Why do you need to create a `ContentItem` for the original NITF file? Would it be simpler to just update the process method to just need the `ByteSource`/`InputStream` and the `Metacard` instead?

Reply:
I was trying to change as little as possible in the processing code. I ended up changing more than I had originally intended so I might as well ditch the ContentItem here.